sympy/matrices/matrices.py

```diff
@@ -1581,6 +1583,12 @@ def _matrix_pow_by_recursion(self, num):
     def __pow__(self, num):
         if not self.is_square:
             raise NonSquareMatrixError()
+        num = sympify(num)
+        if not num.is_integer and (num >= 0) == True and not num % 1 == 0:
```

Comment:
I don't understand these conditions. I think that `__pow__` should only decide which method to use. When the Jordan block method is used, it can find out whether the power can be evaluated or not after computing the blocks.